src/libsyntax/ext/expand.rs

```diff
@@ -1504,7 +1489,8 @@ impl<'a, 'b> Folder for InvocationCollector<'a, 'b> {
         };
 
         if attr.is_some() || !traits.is_empty()  {
-            if !self.cx.ecfg.macros_in_extern_enabled() {
+            if !self.cx.ecfg.macros_in_extern_enabled() &&
+               !self.cx.ecfg.custom_attribute_enabled() {
```

Comment:
I wasn't quite able to discern why this change was needed, you can probably help me out though?

Reply:
Macros on foreign items are feature gated, and in this place custom attributes look like macro attrs, so without this condition they start requiring `feature(macros_in_extern)` as well.
I should probably move feature gate checking for `feature(macros_in_extern)` to some other place in which attribute resolution is already known and we can gate macros, but not other attributes.

Reply:
Ok sounds good to me!